File: Day 3/day_3.py
f = open("./input.txt", 'r')
map = f.readlines()
f.close()

x = [0, 0, 0, 0, 0]         # x position of the toboggan
add = [1, 3, 5, 7, 1]       # The slope (on x)
trees = [0, 0, 0, 0, 0]     # Counter of trees

for i in range(1, len(map)):
    map[i] = map[i].replace("\n", '')
    for j in range(len(add)):
        if(i % 2 == 0 or j != 4):   # on the last slope, the toboggan goes 2 down at time, so if it's on an odd y, skip
            x[j] = (x[j] + add[j]) % (len(map[i]))
            if(map[i][x[j]] == '#'):
                trees[j] += 1

mul = 1
for t in trees:
    mul *= t

print(f"Part 1: {trees[1]}\nPart 2: {mul}")

File: Day 3/day_3.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <vector>

using namespace std;

int main()
{
    fstream f;
    f.open("input.txt");
    string l;
    vector<string> map;

    int x[5] = {0, 0, 0, 0, 0};     // x position of the toboggan
    int add[5] = {1, 3, 5, 7, 1};   // The slope (on x)
    int trees[5] = {0, 0, 0, 0, 0}; // Counter of trees
    int i = 0;

    while (getline(f, l))
    {
        if (i == 0)
        {
            i++;
            continue;
        }
        for (int j = 0; j < 5; j++)
        {
            if (i % 2 == 0 || j != 4) // On the last slope, the toboggan goes 2 down at time, so if it's on an odd y, skip
            {
                x[j] = (x[j] + add[j]) % (l.size());
                if (l[x[j]] == '#')
                {
                    trees[j]++;
                }
            }
        }
        i++;
    }

    unsigned int mul = 1; // signed int overflows, so I used an unsigned int.
    for (int i = 0; i < 5; i++)
    {
        cout << trees[i] << endl;
        mul *= trees[i];
    }

    cout << fixed << "Part 1: " << trees[1] << "\nPart 2: " << mul << endl;

    return 0;
}
